satpy/readers/__init__.py

```diff
@@ -557,31 +557,24 @@ def load_readers(filenames=None, reader=None, reader_kwargs=None):
     reader, filenames, remaining_filenames = _get_reader_and_filenames(reader, filenames)
     (reader_kwargs, reader_kwargs_without_filter) = _get_reader_kwargs(reader, reader_kwargs)
 
-    for idx, reader_configs in enumerate(configs_for_reader(reader)):
-        if isinstance(filenames, dict):
-            readers_files = set(filenames[reader[idx]])
-        else:
-            readers_files = remaining_filenames
+    if reader_kwargs is None:
+        reader_kwargs = {}
```

Comment:
Did something change? why do we need this now?

Reply:
It is needed because of `**reader_kwargs` that I needed for refactoring to make CodeScene happy about the cyclomatic complexity.